src/services/ApiService/OpenSea/OpenSea.ts

```diff
@@ -0,0 +1,62 @@
+import { AxiosInstance } from 'axios';
+
+import { OPENSEA_API } from '@config';
+import { ApiService } from '@services/ApiService';
+
+import { OpenSeaCollection, OpenSeaNFT } from './types';
+
+const NFT_LIMIT_PER_QUERY = 50; // Max allowed by OpenSea
+const NFT_LIMIT_MAX = 1000; // To prevent overly spamming the API - @todo Discuss
+const COLLECTION_LIMIT = 300; // @todo Discuss
+
+export default abstract class OpenSeaService {
```

Comment:
Can you do a factory function instead of a class ?

Reply:
This is the way we create our other services. I don't quite understand your recommendation.

Reply:
I thought we wanted to avoid using classes for new services we created, I remeber this from reviews on my PRs. I don't have a strong opinion on this appart from personal preferencies.

Reply:
Can you point me to an example in the codebase?

Reply:
https://github.com/MyCryptoHQ/MyCrypto/blob/212bb5bd7eae26dd60eb0c9ae0b964409976e2bf/src/services/ApiService/Rates/Rates.ts